### PR TITLE
[subs] bypass directory cache to ensure we're picking up subtitles

### PR DESCRIPTION
**Pull request by mkortstiege**
Should fix picking up external subtitles. cc @arnova, @ace20022 
http://trac.kodi.tv/ticket/15989

### CODE REVIEW

**Comment by ace20022** [comment]
Thanks, +1

File: xbmc/Util.cpp [hunk 1]
```diff
@@ -1754,7 +1754,7 @@ void CUtil::ScanForExternalSubtitles(const std::string& strMovie, std::vector<st
   CFileItemList items;
   vector<std::string> strLookInPaths;
 
-  int flags = DIR_FLAG_NO_FILE_DIRS | DIR_FLAG_READ_CACHE | DIR_FLAG_NO_FILE_INFO;
+  int flags = DIR_FLAG_NO_FILE_DIRS | DIR_FLAG_BYPASS_CACHE | DIR_FLAG_NO_FILE_INFO;
```
Comment (arnova) [inline xbmc/Util.cpp]: DIR_FLAG_BYPASS_CACHE will not update the dir-cache, for which there is no reason. Just drop the DIR_FLAG_READ_CACHE and all should be well.

Comment (mkortstiege) [inline xbmc/Util.cpp]: You're sure that dropping will fix the scenario you described where the user enters the directory and then inserts a subtitle? Bypassing it makes sure we`re re-fetching the directory.

Comment (ace20022) [inline xbmc/Util.cpp]: I tested with the flag removed some days ago.

**Comment by mkortstiege** [comment]
OK: Updated.

**Comment by arnova** [comment]
Thanks. And yes I'm sure this is correct. Only problem I'm seeing is the performance hit with remote filesystems. Perhaps we could add some intelligence where we only fetch the remote dir when we don't have a sub for it already in the dir cache?

**Comment by arnova** [comment]
I see no problem with this.

**Comment by mkortstiege** [comment]
I don't think we should over-engineer this. If it's a real performance issue we can think about it again.

**Comment by arnova** [comment]
Sounds good. I think we previously also had this problem then, so I'm fine with this.

**Comment by mkortstiege** [comment]
Jep. @MartijnKaijser your button.

**Pull request merged**